modin/pandas/concat.py

```diff
@@ -132,3 +139,25 @@ def concat(
         else:
             result_df.columns = new_idx
     return result_df
+
+
+def _determine_name(objs: list, axis):
+    axis_getter = (lambda df: df.columns) if axis else (lambda df: df.index)
+    name_getter = (
+        (lambda df: axis_getter(df).names)
+        if isinstance(axis_getter(objs[0]), pandas.MultiIndex)
+        else (lambda df: axis_getter(df).name)
+    )
+
+    names = [name_getter(obj) for obj in objs]
+
+    if all(
```

Comment:
Why do we need this check? Can you add a comment for the case here as well?

Reply:
pandas saves index name after concatenation only if index names along passed axis are the same for all passed `objs`, so this is what this check are doing

added a comment in the last commit about it